internal/dataplane/parser/translators/ingress.go

```diff
@@ -388,3 +389,109 @@ func MaybePrependRegexPrefixForIngressV1Fn(ingress *netv1.Ingress, applyLegacyHe
 		return lo.ToPtr(MaybePrependRegexPrefix(path, regexPrefix, applyLegacyHeuristic))
 	}
 }
+
+type runeType int
+
+const (
+	runeTypeEscape runeType = iota
+	runeTypeMark
+	runeTypeDigit
+	runeTypePlain
+)
+
+// generateRewriteURIConfig parses uri with SM of four states.
+// `runeTypeEscape` indicates `\` encountered and `$` expected, the SM state
+// will transfer to `runeTypePlain`.
+// `runeTypeMark` indicates `$` encountered and one or more digits expected,
+// the SM state will transfer to `runeTypeDigit`.
+// `runeTypeDigit` indicates at least one digit encountered. If the type of following
+// character is still digit, the SM state will remain unchanged. Otherwise, a new
+// capture group will be created and the SM state will transfer to `runeTypePlain`.
+// `runeTypePlain` indicates the following character is plain text other than `$` and `\`.
+// The former will cause the SM state to transfer to `runeTypeMark` and the latter will
+// cause the SM state to transfer to `runeTypeEscape`.
+func generateRewriteURIConfig(uri string) (string, error) {
```

Comment:
> * Escaping with `$` seems more intuitive, I will change it .
> * Full match capture group is allowed, whether I should forbid it ?
> * Whether I should take capture groups greater than 9 into consideration ?

- What does "Full match capture group" mean here?
- I think we do not need to consider the limit of 9 capture groups. This does not bring much convenience on implementing,  IMO.

Reply:
@rodman10  After some internal discussion, we decided to keep the same behavior as the PR does. So resolve it for now.

Reply:
I have wrapped the capture group index with braces in the form of ${1} so that the situation of digit following capture group index can be dealed properly. @randmonkey